### PR TITLE
docs: update LSP0 Natspec

### PR DESCRIPTION
**Pull request by b00ste**
# What does this PR introduce?

> 🕐 **This PR is in draft and awaiting an update in the `@b00ste/hardhat-dodoc` package.** Do not review yet until this notice is gone and the PR in in "ready for review" mode.

## 📄 Documentation

This PR updates the Natspec of the LSP0 contracts.

### PR Checklist

- [ ] Wrote Tests
- [ ] Wrote & Generated Documentation (readme/natspec/dodoc)
- [ ] Ran `npm run lint` && `npm run lint:solidity` (solhint)
- [ ] Ran `npm run format` (prettier)
- [ ] Ran `npm run build`
- [ ] Ran `npm run test`

### CODE REVIEW

File: contracts/LSP0ERC725Account/LSP0ERC725Account.sol [hunk 1]
```diff
@@ -13,39 +13,24 @@ import {
...
- *
- * - Secure ownership management using LSP14
- *   https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-14-Ownable2Step.md
- *
- * - Extending the account with new functions and interfaceIds of future standards using LSP17
- *   https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-17-ContractExtension.md
- *
- * - Verifying calls on the owner to allow unified and standard interaction with the account using LSP20
- *   https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-20-CallVerification.md
+ * - Detecting supported standards using [ERC-165]
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725Account.sol:16]: I suggest it is not needed to repeat this in the standard and proxy version of the contract. Just having this described in the `Core` is fine.

You can specify in the `@title` for these that it is the standard or proxy or abstract proxy version.

File: contracts/LSP0ERC725Account/ILSP0ERC725Account.sol [hunk 2]
```diff
@@ -21,48 +21,28 @@ import {ILSP14Ownable2Step} from "../LSP14Ownable2Step/ILSP14Ownable2Step.sol";
...
 
     /**
-     * @dev Allows a caller to batch different function calls in one call.
-     * Perform a delegatecall on self, to call different functions with preserving the context
-     * It is not possible to send value along the functions call due to the use of delegatecall.
-     *
+     * @dev Allows a caller to batch different function calls in one call. Perform a `delegatecall` on self, to call different functions with preserving the context.
      * @param data An array of ABI encoded function calls to be called on the contract.
-     * @return results An array of values returned by the executed functions.
+     * @return results An array of abi-encoded returned data by the executed functions.
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/ILSP0ERC725Account.sol:45]:
```suggestion
     * @return results An array of abi-encode data returned by the functions executed.
```

File: contracts/LSP0ERC725Account/ILSP0ERC725Account.sol [hunk 3]
```diff
@@ -21,48 +21,28 @@ import {ILSP14Ownable2Step} from "../LSP14Ownable2Step/ILSP14Ownable2Step.sol";
...
+ * - Extending the account with new functions and interfaceIds of future standards using [LSP-17-ContractExtension]
+ * - Verifying calls on the owner to allow unified and standard interaction with the account using [LSP-20-CallVerification]
  */
 interface ILSP0ERC725Account {
     /**
-     * @notice Emitted when receiving native tokens
-     * @param sender The address of the sender
-     * @param value The amount of native tokens received
+     * @dev Emitted when receiving native tokens.
+     * @notice `value` tokens received from `sender`.
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/ILSP0ERC725Account.sol:36]:
```suggestion
     * @notice `value` native tokens received from `sender`.
```

File: contracts/LSP0ERC725Account/ILSP0ERC725Account.sol [hunk 4]
```diff
@@ -21,48 +21,28 @@ import {ILSP14Ownable2Step} from "../LSP14Ownable2Step/ILSP14Ownable2Step.sol";
...
+     * @param sender The address of the sender.
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/ILSP0ERC725Account.sol:37]:
```suggestion
     * @param sender The address that sent some native tokens to this contract.
```

File: contracts/LSP0ERC725Account/LSP0ERC725Account.sol [hunk 5]
```diff
@@ -13,39 +13,24 @@ import {
...
+ * - Secure ownership management using [LSP-14-Ownable2Step]
+ * - Extending the account with new functions and interfaceIds of future standards using [LSP-17-ContractExtension]
+ * - Verifying calls on the owner to allow unified and standard interaction with the account using [LSP-20-CallVerification]
  */
 contract LSP0ERC725Account is LSP0ERC725AccountCore {
     /**
-     * @notice Deploying the contract with owner set to: `initialOwner`
-     * @dev Set `initialOwner` as the contract owner.
-     * The `constructor` also allows funding the contract on deployment.
+     * @notice Deploying an LSP0ERC725Account with owner set to: `initialOwner`
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725Account.sol:27]:
```suggestion
     * @notice Deploying a LSP0ERC725Account contract with owner set to address `initialOwner`
```

File: contracts/LSP0ERC725Account/LSP0ERC725Account.sol [hunk 6]
```diff
@@ -13,39 +13,24 @@ import {
...
+     * @dev Set `initialOwner` as the contract owner. The `constructor` also allows funding the contract on deployment.
      * @param initialOwner The owner of the contract.
      *
-     * @custom:events {ValueReceived} event when funding the contract on deployment.
+     * @custom:events
+     * - {ValueReceived} event when funding the contract on deployment.
+     * - {OwnershipTransferred} event when `initialOwner` is set.
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725Account.sol:33]:
```suggestion
     * - {OwnershipTransferred} event when `initialOwner` is set as the contract {owner}.
```

File: docs/contracts/ERC725/ERC725.md
Comment (CJ42) [inline docs/contracts/ERC725/ERC725.md:1]: Nice 👌

**Review by CJ42 — CHANGES_REQUESTED** [review]
Added first round of review.

File: contracts/LSP0ERC725Account/ILSP0ERC725Account.sol [hunk 7]
```diff
@@ -16,53 +16,23 @@ import {
...
+     * @param value The amount of native tokens received.
      */
     event ValueReceived(address indexed sender, uint256 indexed value);
 
     /**
-     * @dev Allows a caller to batch different function calls in one call.
-     * Perform a delegatecall on self, to call different functions with preserving the context
-     * It is not possible to send value along the functions call due to the use of delegatecall.
-     *
+     * @dev Allows a caller to batch different function calls in one call. Perform a `delegatecall` on self, to call different functions with preserving the context.
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/ILSP0ERC725Account.sol:33]: Missing `@notice` tag

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol [hunk 8]
```diff
@@ -68,34 +74,18 @@ import {
...
- *   https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-14-Ownable2Step.md
- *
- * - Extending the account with new functions and interfaceIds of future standards using LSP17
- *   https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-17-ContractExtension.md
- *
- * - Verifying calls on the owner to allow unified and standard interaction with the account using LSP20
- *   https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-20-CallVerification.md
+ * - Detecting supported standards using [ERC-165]
+ * - Executing several operation on other addresses including creating contracts using [ERC-725X]
+ * - Storing data in a generic way using [ERC-725Y]
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:83]:
```suggestion
 * - A generic data key-value store using [ERC-725Y]
```

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol [hunk 9]
```diff
@@ -68,34 +74,18 @@ import {
...
+ * - Validating signatures using [ERC-1271]
+ * - Receiving notification and react on them using [LSP-1-UniversalReceiver]
+ * - Secure ownership management using [LSP-14-Ownable2Step]
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:86]:
```suggestion
 * - Safer ownership management through 2-steps transfer using [LSP-14-Ownable2Step]
```

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol [hunk 10]
```diff
@@ -68,34 +74,18 @@ import {
...
+ * - Extending the account with new functions and interfaceIds of future standards using [LSP-17-ContractExtension]
+ * - Verifying calls on the owner to allow unified and standard interaction with the account using [LSP-20-CallVerification]
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:88]:
```suggestion
 * - Verifying calls on the owner to make it easier to interact with the account directly using [LSP-20-CallVerification]
```

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol [hunk 11]
```diff
@@ -127,28 +117,20 @@ abstract contract LSP0ERC725AccountCore is
     // solhint-disable no-complex-fallback
 
     /**
-     * @notice Achieves the goal of [LSP17-ContractExtension] standard by extending the contract to
-     * handle calls of functions that do not exist natively, forwarding the function call to the
-     * extension address mapped to the function being called.
+     * @notice Achieves the goal of [LSP-17-ContractExtension] standard by extending the contract to handle calls of functions that do not exist natively, forwarding the function call to the extension address mapped to the function being called.
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:120]: This is not a `@notice` tag to display to the user. Suggesting to refactor and put the `@notice` tag in the interface if possible.

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol [hunk 12]
```diff
@@ -167,12 +149,9 @@ abstract contract LSP0ERC725AccountCore is
...
 
     /**
-     * @dev Allows a caller to batch different function calls in one call.
-     * Perform a delegatecall on self, to call different functions with preserving the context
-     * It is not possible to send value along the functions call due to the use of delegatecall.
+     * @inheritdoc ILSP0ERC725Account
      *
-     * @param data An array of ABI encoded function calls to be called on the contract.
-     * @return results An array of values returned by the executed functions.
+     * @custom:info It's not possible to send value along the functions call due to the use of `delegatecall`.
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:154]: Nice use of callouts 👌

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol [hunk 13]
```diff
@@ -263,24 +237,18 @@ abstract contract LSP0ERC725AccountCore is
...
+     * @inheritdoc IERC725X
      *
      * @custom:requirements
-     * - The length of the parameters provided MUST be equal
-     * - if a `value` is provided, the contract MUST have at least this amount in its balance to execute successfully.
-     * - if the operation type is `STATICCALL` or `DELEGATECALL`, `value` SHOULD be 0.
-     * - `target` SHOULD be `address(0)` when deploying a contract.
-     * - MUST pass when called by the owner or by an authorised address that pass the verification check performed
-     * on the owner accordinng to [LSP20-CallVerification] specification
+     * - The length of the parameters provided must be equal.
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:243]: Nice, let's keep these in this repo and not in ERC725 (So ERC725 does not have any `@custom` tag and is not tied to our docs.

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol [hunk 14]
```diff
@@ -430,25 +390,22 @@ abstract contract LSP0ERC725AccountCore is
     }
 
     /**
-     * @notice Achieves the goal of [LSP-1-UniversalReceiver] by allowing the account to be notified about incoming/outgoing
-     * transactions and enabling reactions to these actions.
+     * @notice Achieves the goal of [LSP-1-UniversalReceiver] by allowing the account to be notified about incoming/outgoing transactions and enabling reactions to these actions.
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:393]: Suggest to review this `@notice` tag too.

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol [hunk 15]
```diff
@@ -690,14 +628,14 @@ abstract contract LSP0ERC725AccountCore is
     }
 
     /**
-     * @notice Achieves the goal of ERC165 to detect supported interfaces and LSP17 by
+     * @notice Achieves the goal of [ERC-165] to detect supported interfaces and [LSP-17-ContractExtension] by
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:631]: Same here.

**Review by CJ42 — CHANGES_REQUESTED** [review]
Added some requested changes.

Also suggesting to re-use the `@notice` tags from ERC725 when possible, and make sure they are there :)

**Review by YamenMerhi — APPROVED** [review]

**Review by CJ42 — APPROVED** [review]

**Pull request merged**